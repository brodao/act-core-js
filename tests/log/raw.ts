import actLogger from '../../src/logger';

const { wrap } = require('@brodao/act-jest-snapshot-console');

beforeEach(() => {
	actLogger.config.raw = true;
});

afterEach(() => {
	actLogger.config.raw = false;
});

describe('Console em texto puro', () => {
	test("Apresenta o 'splash'", () => {
		expect(
			wrap(() =>
				actLogger.showBanner({
					name: 'test_show_banner',
					version: '99.99.99',
					description: 'Show Banner',
				})
			)
		).toMatchSnapshot();
	});

	test('Aviso simples', () => {
		expect(wrap(() => actLogger.warn('Aviso simples'))).toMatchSnapshot();
	});

	test('Aviso com vários agumentos', () => {
		expect(
			wrap(() =>
				actLogger.warn('Aviso com argumentos', 'arg 1', 'arg 2', 'arg 3')
			)
		).toMatchSnapshot();
	});

	test('Erro simples', () => {
		expect(wrap(() => actLogger.error('Erro simples'))).toMatchSnapshot();
	});

	test('Erro com vários agumentos', () => {
		expect(
			wrap(() =>
				actLogger.error('Erro com argumentos', 'arg 1', 'arg 2', 'arg 3')
			)
		).toMatchSnapshot();
	});
});
