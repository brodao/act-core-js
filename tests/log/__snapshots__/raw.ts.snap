// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Console em texto puro Apresenta o 'splash' 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "L:/====================v======================================================\\\\
L:|     /////// ////// | AC TOOLS - Extensions for VS-Code and NodeJS   |
L:|    //   // //      | (C) 2020 Alan Candido (BRODAO) <brodao@gmail.com>    |
L:|   /////// //       |                                                      |
L:|  //   // //        | Support tools for TOTVS developers         |
L:| //   // //////     | https://github.com/brodao/workspace/projects/AFPV    |
L:\\\\====================^======================================================/
L:test_show_banner [99.99.99] Show Banner
",
}
`;

exports[`Console em texto puro Aviso com vários agumentos 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "L:Aviso com argumentos
",
}
`;

exports[`Console em texto puro Aviso simples 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "L:Aviso simples
",
}
`;

exports[`Console em texto puro Erro com vários agumentos 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "L:Erro com argumentos
",
}
`;

exports[`Console em texto puro Erro simples 1`] = `
Object {
  Symbol(act-jest-snapshot-raw): undefined,
  Symbol(act-jest-snapshot-console): "L:Erro simples
",
}
`;
